extensions/scarb-test-collector/src/main.rs

```diff
@@ -0,0 +1,45 @@
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use std::str::FromStr;
+
+use scarb_metadata::MetadataCommand;
+
+use crate::compilation::compile_tests;
+use crate::crate_collection::collect_test_compilation_targets;
+use crate::metadata::{
+    corelib_for_package, dependencies_for_package, id_for_package, paths_for_package,
+};
+
+mod compilation;
+mod crate_collection;
+mod metadata;
+
+#[derive(Parser)]
+#[command(version)]
```

Comment:
Add a help line saying that this is Starknet Foundry private extension and user should not call it manually.